vpr/src/route/gen/rr_graph_uxsdcxx.h

```diff
@@ -35,6 +35,11 @@ namespace uxsd {
  */
 inline void get_line_number(const char* filename, std::ptrdiff_t offset, int* line, int* col);
 
+[[noreturn]] inline void noreturn_report(const std::function<void(const char*)>* report_error, const char* msg) {
+    (*report_error)(msg);
+    throw std::runtime_error("Unreachable!");
+}
+
```

Comment:
I'm a little concerned about this, as the error reporting was already rather confusing in this code, and I think having another error reporting method does not help. There are now **4** different error reporting methods in the RR graph reader: 
- noreturn_report() member function
- report_error_ std::function
- report_error() member function
- error_encountered()


Additionally, the way we've generally handled this in VPR in the past, is to just ensure that such variables are always initialized (even if the code path where the value may be used may seem impossible). Its also generally more robust (e.g. across compilers with differing ability/choices with respect to reasoning about uninitialized values).

Reply:
The ultimate fix was to `memset` the variables:
https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1120/files#diff-869c3d00310fe12fba0faea5ab40ddccL2795-L2798

The `noreturn_report` function is required to annotated the "noreturn" of the std::function.  As far as I can tell, you cannot mark a function pointer as noreturn.

Reply:
Ah now I see. I also thought this was in the non-auto generated rr_graph_uxsdcxx_serializer.h, but I'm much less concerned about the extra layer of error reporting if it's in the auto-generated code.